packages/jupyter-ai/jupyter_ai/models.py

```diff
@@ -80,3 +82,23 @@ class DescribeTaskResponse(BaseModel):
 class ChatHistory(BaseModel):
     """History of chat messages"""
     messages: List[ChatMessage]
+
+
+class ListProvidersEntry(BaseModel):
+    """Model provider with supported models
+    and provider's authentication strategy
+    """
+    id: str
+    name: str
+    models: List[str]
+    auth_strategy: AuthStrategy
+
+
+class ListProvidersResponse(BaseModel):
+    providers: List[ListProvidersEntry]
+
+
+class ProviderConfig(BaseModel):
+    model_provider: str
+    embeddings_provider: str
+    api_keys: Dict[str, str]
```

Comment:
Few comments on this model:

1. When we are referencing the ID of a provider, let's always make it explicit and always suffix with `_id`. So here, the keys should be `model_provider_id` and `embedding_provider_id`. 

2. I'm not sure about naming the model `ProviderConfig` (and similarly for the handlers); it seems more apt to name this `GlobalConfig`, since configuration will not necessarily be limited to providers in the future.

3. This should probably be split into two objects:

- `GetGlobalConfigResponse`: same as current model
- `UpdateGlobalConfigRequest`: same as current model, but all keys are optional

Reply:
GlobalConfig seems overly broad. How about we just call it Config?

Reply:
How about `ChatConfig`?

Reply:
Let me track this in #129 actually.